File: src/main/kotlin/solutionsJava/MissingNumber.java
import java.util.Arrays;
import java.util.Scanner;

public class MissingNumber {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        scanner.nextLine();
        int[] numbers = new int[n - 1];
        String[] numbersLine = scanner.nextLine().split(" ");

        for (int i = 0; i < numbers.length; i++) {
            numbers[i] = Integer.parseInt(numbersLine[i]);
        }

        System.out.println(missingNumber(numbers));
    }

    // sort the array, and find a missing number
    private static int missingNumber(int[] numbers) {
        int[] sorted = numbers.clone();
        Arrays.sort(sorted);

        if (sorted[0] != 1) return 1;

        for (int i = 1; i < sorted.length; i++) {
            if (sorted[i] - sorted[i - 1] != 1) {
                return sorted[i] - 1;
            }
        }

        return numbers.length + 1;
    }
}

File: src/main/kotlin/solutions/MissingNumber.kt
package uz.safix.solutions

import java.util.Scanner

fun main() {
    val scanner = Scanner(System.`in`)
    val n = scanner.nextInt(); scanner.nextLine()
    val numbers = IntArray(n - 1)
    val numbersLine = scanner.nextLine().split(" ")

    for (i in numbers.indices) { numbers[i] = numbersLine[i].toInt() }

    println(missingNumber(numbers))
}

// sort the array, and find a missing number
private fun missingNumber(numbers: IntArray): Int {
    val sorted = numbers.sorted()
    if (sorted[0] != 1) return 1

    for (i in 1 until sorted.size) {
        if ((sorted[i] - sorted[i - 1] != 1)) {
            return sorted[i] - 1
        }
    }

    return numbers.size + 1
}